src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -306,60 +845,142 @@ def map_keyvalue_to_featureflagvalue(keyvalue):
             for client_filter in client_filters:
                 # If there is a filter, it should always have a name
                 # In case it doesn't, ignore this filter
-                name = client_filter.get(FeatureFlagConstants.FILTER_NAME)
+                lowercase_filter = {k.lower(): v for k, v in client_filter.items()}
+                name = lowercase_filter.get(FeatureFlagConstants.NAME)
                 if name:
-                    params = client_filter.get(FeatureFlagConstants.FILTER_PARAMETERS, {})
+                    params = lowercase_filter.get(
+                        FeatureFlagConstants.FILTER_PARAMETERS, {}
+                    )
                     client_filters_list.append(FeatureFilter(name, params))
                 else:
-                    logger.warning("Ignoring this filter without the %s attribute:\n%s",
-                                   FeatureFlagConstants.FILTER_NAME,
-                                   json.dumps(client_filter, indent=2, ensure_ascii=False))
+                    raise ValidationError(
```

Comment:
Just to confirm, we are breaking on an invalid filter instead of ignoring the filter, right?

Reply:
We should still be ignoring the filter because if throwing on an invalid filter would be a breaking change